src/realm/object-store/c_api/sync.cpp

```diff
@@ -404,7 +405,7 @@ RLM_API void realm_sync_config_set_after_client_reset_handler(realm_sync_config_
         realm_t r1{before_realm};
         auto tsr = realm_t::thread_safe_reference(std::move(after_realm));
         if (!callback(userdata.get(), &r1, &tsr, did_recover)) {
-            throw CallbackFailed();
+            throw CallbackFailed{ErrorStorage::get_thread_local()->get_and_clear_usercode_error()};
```

Comment:
Do we need to capture this here because some other callback might overwrite it before we get to the sync error handler? Why not just add a `usercode_error` to `realm_sync_error_t` and set it from `ErrorStorage::get_thread_local()->get_and_clear_usercode_error()` when we're in the sync error handler?

Reply:
`realm_register_user_code_callback_error` can only be called by the SDK which is basically registering the exception ptr.  So it cannot be overwritten in any way by other callbacks.

Reply:
Right, my point is that by this point either the SDK has called realm_register_user_code_callback_error or it hasn't and if it has we can get that usercode_error from where we handle sync errors in general out in the C API.